### PR TITLE
API restructuring, demo updated & replace_explicit draft

### PR DESCRIPTION
**Pull request by Zethson**
Part of #45 

1. This PR includes API shortcuts like pp for processing, ec for encode etc
2. I've updated our demo notebook to accommodate the changes in this PR
3. This PR adds a draft for explicit replacement of missing values

The replace_explicit function is not yet done nor has it tests!
However, I would like to include them in a follow up PR as soon as I find the time.

Due to the restructuring I would like to merge this one beforehand.

### CODE REVIEW

File: ehrapy/api/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from ehrapy.api import data as dt
```
Comment (Zethson) [inline ehrapy/api/__init__.py:1]: Does not include IO here because we are not renaming it. Should be good.

File: ehrapy/api/preprocessing/__init__.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    Returns:
+        :class:`~anndata.AnnData` object with imputed X
+
+    Example:
+        .. code-block:: python
+
+            import ehrapy.api as ep
+            adata = ep.data.mimic_2(encode=True)
+            adata_replaced = ep.pp.replace_explicit(adata_3, replacement=0, copy=True)
```
Comment (Imipenem) [inline ehrapy/api/preprocessing/__init__.py:31]: does this work? what is adata_3 here?

Comment (Zethson) [reply ehrapy/api/preprocessing/__init__.py:31]: a copy pasta mistake from my side

File: ehrapy/api/preprocessing/_data_imputation.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        Returns:
+            :class:`~anndata.AnnData` object with imputed X
+        """
+        adata_to_act_on = adata
+        if copy:
+            adata_copy = adata.copy()
+            adata_to_act_on = adata_copy
+
+        # scenario 1: Replace all missing values with the specified value
+        adata_to_act_on.X[np.isnan(adata_to_act_on.X)] = replacement
```
Comment (Imipenem) [inline ehrapy/api/preprocessing/_data_imputation.py:38]: does `isnan()` also work on empty string for example?

File: ehrapy/api/preprocessing/_data_imputation.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from typing import Dict, Optional, Tuple, Union
+
+import numpy as np
+from anndata import AnnData
+
+
+class Imputation:
```
Comment (Imipenem) [inline ehrapy/api/preprocessing/_data_imputation.py:7]: In general; It would be better to no not have this in this PR, since it's not fully implemented and I really did not expected it here, since this would be something that better fits into quality control.

File: pyproject.toml [hunk 5]
```diff
@@ -33,6 +33,7 @@ scikit-learn = "^1.0"
 category_encoders = "^2.2.2"
 scanpy = "^1.8.1"
 mudata = "^0.1.0"
+leidenalg = "^0.8.7"
```
Comment (Imipenem) [inline pyproject.toml:36]: why is this added?

File: ehrapy/api/preprocessing/_data_imputation.py (hunk 4 above)
Comment (Zethson) [reply ehrapy/api/preprocessing/_data_imputation.py:7]: I needed it for the notebook update. But I can add it in another PR that you can then use as your base.

File: pyproject.toml (hunk 5 above)
Comment (Zethson) [reply pyproject.toml:36]: Leidenalg is an optional dependency in Scanpy. However, we need it for clustering as is now included in our example notebook. It should imo always be installed.

File: ehrapy/api/preprocessing/_data_imputation.py (hunk 4 above)
Comment (Imipenem) [reply ehrapy/api/preprocessing/_data_imputation.py:7]: IMO, for now its fine, just wanted to state it.

**Comment by Imipenem** [comment]
Before merging this, tests should be added here
